### ISSUE TITLE
Add aspect selector

### ISSUE THREAD

**Issue by oleksandr-pavlyk** [IntelPython/dpctl#508]
There is ``sycl::aspect_selector``, which allows to select devices based on a collection of requested aspects, as well as on the list of rejected aspects. See https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#table.device.selectors

Perhaps this calls for a similar `dpctl.select_device_with_aspects`.

**Comment by oleksandr-pavlyk** [comment]
Aspect selector in C++ requires specifying required present and required absent aspects to be given at the compile time.

In Python this will be done dynamically. Aspects need to be encoded with strings.

The list of device aspects in SYCL is given in [section 4.6.4.3](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device-aspects). 

```c++
namespace sycl {

enum class aspect {
  cpu,
  gpu,
  accelerator,
  custom,
  emulated,
  host_debuggable,
  fp16,
  fp64,
  atomic64,
  image,
  online_compiler,
  online_linker,
  queue_profiling,
  usm_device_allocations,
  usm_host_allocations,
  usm_atomic_host_allocations,
  usm_shared_allocations,
  usm_atomic_shared_allocations,
  usm_system_allocations
};

} // namespace sycl
```

Perhaps these could be given as strings. The function can be based on [example/python/device_selection.py](https://github.com/IntelPython/dpctl/blob/master/examples/python/device_selection.py#L79).

**Comment by oleksandr-pavlyk** [comment]
Also see #309

**Comment by oleksandr-pavlyk** [comment]
This was implemented.

**Comment by oleksandr-pavlyk** [comment]
Implemented with #558
